src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java

```diff
@@ -462,57 +467,58 @@ public String publishSpeechRequested(String toSpeak) {
    * @param block
    * @return
    */
-  public List<AudioData> parse(String toSpeak, boolean block){
-    
+  public List<AudioData> parse(String toSpeak, boolean block) {
+
     // TODO - not sure if we want to support this notation
     // but at the moment it seems useful 
     // splitting on sound effects ...
     // TODO - use SSML speech synthesis markup language
-    
+
     // broadcast the original text to be processed/parsed
     invoke("publishSpeechRequested", toSpeak);
-    
+
     // normalize to lower case
     toSpeak = toSpeak.toLowerCase();
-    
+
     // process substitutions
-    for (String substitute: substitutions.keySet()) {
+    for (String substitute : substitutions.keySet()) {
       toSpeak = toSpeak.replace(substitute, substitutions.get(substitute));
     }
 
     List<String> spokenParts = parseEffects(toSpeak);
-    
+
     toSpeak = filterText(toSpeak);
-    
+
     // we generate a list of audio data to play to support
     // synthesizing this speech
     List<AudioData> playList = new ArrayList<AudioData>();
-   
+
     for (String speak : spokenParts) {
-  
+
       AudioData audioData = null;
       if (speak.startsWith("#") && speak.endsWith("#")) {
-        audioData = new AudioData(System.getProperty("user.dir") + File.separator + "audioFile" + File.separator + "voiceEffects" + File.separator + speak.substring(1, speak.length() - 1) + ".mp3");
+        audioData = new AudioData(
+            System.getProperty("user.dir") + File.separator + "audioFile" + File.separator + "voiceEffects" + File.separator + speak.substring(1, speak.length() - 1) + ".mp3");
```

Comment:
The path for where the cached mp3 files should not be computed here, it should be pulled out as a variable for this class and referenced here.  That should have proper setters / getters on it so that people can configure it as needed.

Reply:
I agree - default could be this, but I agree on the setters & getters in the AbstractSpeechSynthesis class

Reply:
I'd prefer to see this pulled out where there is a member for 

private String mp3CacheDir = System.getProperty("user.dir") + File.separator;     that has a getter/setter

and then when creating this AudioData it would use that .like the following 

new AudioData(mp3CacheDir + "audioFile" + File.separator + "voiceEffects" + File.separator + speak.substring(1, speak.length() - 1) + ".mp3")

I also don't like all the speak.substring stuff.. i'm not sure why the whole utterance isn't just hashed and used ...

Reply:
ok for setters / getters...
about speak.substring it is related to audioeffects only. 
convention to call it is #R2D2# and file is named R2D2.mp3